### PR TITLE
bio-rd BGP configuration refactoring

### PR DESCRIPTION
**Pull request by czerwonk**
This PR tries to improve the bio-rd BGP configuration.
It aims to provide the basic capability to run bio-rd in the current state after checking out supporting most out of the current feature set.

This is work in progress and thus only a draft for now.

In detail:
* adds IPv6 support
* improve readability
* encapsulate BGP server configuration
* allow to change BGP listen address
* respect configured VRF
* add group / neighbor inheritance for AFI specific configurations
* add tests for BGP config loading

### CODE REVIEW

File: cmd/bio-rd/bgp.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+	return nil
+}
+
+func (c *bgpConfigurator) deconfigureRemovedSessions(cfg *config.BGP) {
+	for _, p := range c.srv.GetPeers() {
+		found := false
+		for _, g := range cfg.Groups {
+			for _, n := range g.Neighbors {
+				if n.PeerAddressIP == p.Addr() && p.VRF() == c.srv.GetDefaultVRF() {
```
Comment (BarbarossaTM) [inline cmd/bio-rd/bgp.go:43]: The `p.VRF() == c.srv.GetDefaultVRF()` bit surprises me a bit, is the intention to only support BGP sessions in the main VRF, or is this subject to change in further iterations? :grin:

Comment (czerwonk) [reply cmd/bio-rd/bgp.go:43]: This code is nearly unmodified to the old main.go code, so it's subject to change in a further iteration.
The first step focused on readability of the existing code to set a starting point for the next steps.

Comment (BarbarossaTM) [reply cmd/bio-rd/bgp.go:43]: Cool, thanks for untangling this! <3

**Review by BarbarossaTM — DISMISSED** [review]
Thanks for this! This looks good to me and is a great enhancement for the config management! <3

I think it would be nice to add allow inheriting the iBGP Cluster ID from the BGP group, too.
I'd be happy to add a small patch on top of this to allow it, if you're exhausted after this :-)

File: cmd/bio-rd/config/bgp.go [hunk 2]
```diff
@@ -28,25 +28,26 @@ func (b *BGP) load(localAS uint32, policyOptions *PolicyOptions) error {
...
+	PeerAS               uint32     `yaml:"peer_as"`
+	LocalAS              uint32     `yaml:"local_as"`
+	HoldTime             uint16     `yaml:"hold_time"`
+	Multipath            *Multipath `yaml:"multipath"`
+	Import               []string   `yaml:"import"`
+	ImportFilterChain    filter.Chain
+	Export               []string `yaml:"export"`
+	ExportFilterChain    filter.Chain
+	RouteServerClient    *bool                `yaml:"route_server_client"`
+	RouteReflectorClient *bool                `yaml:"route_reflector_client"`
```
[the remark this answers is not in the document]
Comment (czerwonk) [reply cmd/bio-rd/config/bgp.go]: Nope, just added the ``Cluster ID`` :-)

**Review by BarbarossaTM — APPROVED** [review]
Sweet! :-)

**Pull request merged**